legacy/src/main/java/org/opensearch/sql/legacy/executor/join/HashJoinElasticExecutor.java

```diff
@@ -124,16 +130,26 @@ private List<SearchHit> createCombinedResults(TableInJoinRequestBuilder secondTa
     Integer hintLimit = secondTableRequest.getHintLimit();
     SearchResponse searchResponse;
     boolean finishedScrolling;
+    LocalClusterState clusterState = LocalClusterState.state();
+    Boolean paginationWithSearchAfter =
+        clusterState.getSettingValue(SQL_PAGINATION_API_SEARCH_AFTER);
     if (hintLimit != null && hintLimit < MAX_RESULTS_ON_ONE_FETCH) {
       searchResponse = secondTableRequest.getRequestBuilder().setSize(hintLimit).get();
       finishedScrolling = true;
     } else {
-      searchResponse =
-          secondTableRequest
-              .getRequestBuilder()
-              .setScroll(new TimeValue(60000))
-              .setSize(MAX_RESULTS_ON_ONE_FETCH)
-              .get();
+      SearchRequestBuilder request =
+          secondTableRequest.getRequestBuilder().setSize(MAX_RESULTS_ON_ONE_FETCH);
+
+      if (paginationWithSearchAfter) {
```

Comment:
I am a bit concerned to put this branching logic in many places since it reduces maintainability. Please consider refactoring if possible.